pkg/asset/installconfig/gcp/dns.go

```diff
@@ -34,41 +40,75 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// GetPublicZoneByID returns a DNS managed zone from the provided project which matches the ID.
+func GetPublicZoneByID(ctx context.Context, project, id string) (*dns.ManagedZone, error) {
+	client, err := NewClient(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	dnsZone, err := client.GetPublicDNSZoneByID(ctx, project, id)
+	if err != nil {
+		return nil, err
+	}
+	return dnsZone, nil
+}
+
 // GetBaseDomain returns a base domain chosen from among the project's public DNS zones.
-func GetBaseDomain(project string) (string, error) {
+func GetBaseDomain(project string) (*Zone, error) {
 	client, err := NewClient(context.TODO())
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
 	defer cancel()
 
-	publicZones, err := client.GetPublicDomains(ctx, project)
+	publicZonesMap, err := client.GetPublicDomains(ctx, project)
 	if err != nil {
-		return "", errors.Wrap(err, "could not retrieve base domains")
+		return nil, errors.Wrap(err, "could not retrieve base domains")
 	}
+
+	publicZones := make([]string, 0, len(publicZonesMap))
+	for name := range publicZonesMap {
+		for _, id := range publicZonesMap[name] {
+			publicZones = append(publicZones, fmt.Sprintf("%s (%s)", name, id))
+		}
+	}
+
 	if len(publicZones) == 0 {
-		return "", errors.New("no domain names found in project")
+		return nil, errors.New("no domain names found in project")
 	}
 	sort.Strings(publicZones)
 
-	var domain string
+	var publicZoneNameChoice string
+	var publicZoneName string
+	var publicZoneID string
```

Comment:
nit: predeclaration here is not necessary for `publicZoneName` and `publicZoneID`

Reply:
Fixed.